NOTICE.md

```diff
@@ -0,0 +1,31 @@
+## NOTICES
+

```

Comment:
I'm not certain if we will eventually include non-WDK content or not, but for library organization I do prefer to have a folder/namespace that is for drivers and not just general Windows code so that its purpose is clear.  We could rename it to "drivers" instead of "wdk", perhaps?

Re: KMDF/WDM: at the time I made this I was mostly following the SDV style (which includes WDM, KMDF, NDIS, and storport-specific rules divided into those folders.)  I doubt that we plan to write UMDF queries in the immediate future, so I don't know that we should add a "UMDF" folder.

Thoughts?

Reply:
I don't think we need so many levels in the repo in general, and the distinction between these is unclear. The purpose of the repo is for CodeQL Windows driver development and analysis, and that is mostly inherent in the name.
Level 1: We're probably not going to add any other static analysis tools to this repo in the near future so we can probably remove the "**codeql**" level without any issue, and add it later if there ever comes a need.
Level 2: We have "**codeql-queries**" and "**windows-drivers/queries**" which is essentially "**general queries**" and "**driver specific queries**" (note that using slashes in directory names is not a good practice and introduces further confusion since the slash denotes hierarchy and not a separation of topic; use dashes if you must but they are probably not necessary here)
Level 3: (Under "windows-drivers/queries") we have "**Likely Bugs**" "**Security/Cryptography**" "**Windows/wdk**" "**microsoft/code/cpp/commons**" and "**suites**" which is extremely confusing as there is no clear organization or distinction between these. Is Security/Cryptography a category of rules? If so, wouldn't that belong in "**Suites**"? Is "microsoft/code/cpp/commons" not part of "Windows/wdk"? Perhaps these should move under a directory called "**libraries**" since they are not queries?
Level 4: (Under "Windows/wdk") Is there a material difference in how you approach a KMDF driver query versus a WDM query or a logical way of organizing these queries? In my opinion, the purpose of the distinction is unclear because WDM should be a parent to KMDF, but if we write the queries exactly the same whether they are WDM or WDM/KMDF, then what is the need to make the separation? For example, if we have sibling categories "Apples" and "Green Apples." Unless we're doing something special on "Green Apples" then why separate them from the other apples? That's why I suggested adding the UMDF directory. Because having parent directory "Apples" with sub directories "Green Apples" and "Red Apples" makes more sense to me since we treat "Green Apples" (KMDF queries) and "Red Apples" (UMDF queries) differently.

Reply:
I'm working to reorganize this per your guidelines above, but just to answer a few of your questions:
- The "Level 3" folders and organization are largely because we took several queries and libraries from the internal Microsoft CodeQL repo (Raul and Josh's repo) and were moved that way to avoid breaking any dependencies and to maintain organizational consistency with as they were in that query.  We can move them around, it just may be slightly more work in the future to update them with any changes from Raul/Josh's internal repo.
- Although it's true that WDM can in some ways be thought of a "parent" of KMDF, when validating driver behavior (especially looking forward to porting SDV rules), having a separation in queries and libraries is useful and even necessary.  Behavior that is correct in a WDM context may not be correct in a KMDF context and vice-versa.  There may be some libraries and queries that are universally applicable (particularly in the CA space) but I would prefer to leave "wdm" and "kmdf" as siblings as opposed to a parent/child relationship, with global queries in their parent folder.  If desired we could have a structure where "windows/drivers" is the top level, "wdm" and "wdf" are the children, and "kmdf" and "umdf" are the children of "wdf".